org.eclipse.xtext.xbase/src/org/eclipse/xtext/xbase/compiler/XbaseCompiler.java

```diff
@@ -1912,7 +1912,7 @@ protected void appendClosureParameter(JvmFormalParameter closureParam, Lightweig
 		appendable.append(parameterType);
 		appendable.append(" ");
 		final String proposedParamName = makeJavaIdentifier(closureParam.getName());
-		String name = appendable.declareVariable(closureParam, proposedParamName);
+		String name = appendable.declareUniqueNameVariable(closureParam, proposedParamName);
```

Comment:
have no idea if this is the correct solution

Reply:
It appears to be ok. In the compiled Java code, it'd be possible to refer to outer variables now which would be shadowed in Xbase itself, but that's not a problem.